src/Collection/Iterator/ExtractIterator.php

```diff
@@ -69,4 +71,37 @@ public function current()
 
         return $extractor(parent::current());
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * We perform here some strictness analysis so that the
+     * iterator logic is bypassed entirely.
+     *
+     * @return \Iterator
+     */
+    public function unwrap()
+    {
+        $iterator = $this->getInnerIterator();
+
+        if ($iterator instanceof CollectionInterface) {
+            $iterator = $iterator->unwrap();
+        }
+
+        if (!$iterator instanceof ArrayIterator) {
+            return $this;
+        }
+
+        // ArrayIterator can be traversed strictly.
+        // Let's do that for performance gains
+
+        $callback = $this->_extractor;
+        $res = [];
+
+        foreach ($iterator->getArrayCopy() as $k => $v) {
```

Comment:
I'm guessing that getting an array copy is part of the optimization windfall as you skip all the function calls to the Iterator interface methods?